src/CMakeLists.txt

```diff
@@ -28,54 +24,19 @@ add_definitions(-DUSE_ONLINE_CAMERA_CALIB)
 # rid of accumulated numeric error
 #add_definitions(-DENFORCE_SO3_FREQ=50)
 
-include_directories(
-  ${PROJECT_SOURCE_DIR}
-  ${CMAKE_SOURCE_DIR}/common)
-
-link_directories(
-  ${CMAKE_SOURCE_DIR}/lib)
-
-list(APPEND deps
-        opencv_highgui
-        opencv_features2d
-        opencv_core
-        opencv_video
-        opencv_imgproc
-        opencv_imgcodecs
-        opencv_xfeatures2d
-        glog
-        GLEW
-        pangolin
-        pthread
-        jsoncpp
-        DBoW2
-        libpnp
-        common
-        ceres
-        cxsparse
-        cholmod
-        lapack
-        blas
-        )
-if (IS_LINUX)
-  list(APPEND deps GL)
-endif (IS_LINUX)
-
-if (USE_GPERFTOOLS)
-  list(APPEND deps profiler)
-endif (USE_GPERFTOOLS)
-
-add_library(xapp STATIC
+
+add_library(xapp SHARED
         estimator_process.cpp
         loader.cpp
         geometry.cpp
         metrics.cpp
         publisher.cpp
         graphwriter.cpp
         viewer.cpp)
-target_link_libraries(xapp ${deps})
+target_link_libraries(xapp PUBLIC common)
+
 
-add_library(xest STATIC
+add_library(xest SHARED
```

Comment:
Any particular reason for building a shared library instead of a static library?